e2e/tests/keycloak-openmrs-flows.spec.ts

```diff
@@ -15,18 +15,18 @@ test('Creating an OpenMRS role creates the corresponding Keycloak role.', async
   // setup
   homePage = new HomePage(page);
   await page.goto(`${O3_URL}/openmrs/admin/users/role.list`);
-  await homePage.addOpenMRSRole();
 
   // replay
+  await homePage.addOpenMRSRole();
+
+  // verify
   await homePage.goToKeycloak();
   await expect(page).toHaveURL(/.*console/);
   await homePage.goToClients();
-  await homePage.goToOpenMRSClient();
-
-  // verify
+  await homePage.selectOpenMRSId()
   await expect(page.getByText(`${randomOpenMRSRoleName.roleName}`)).toBeVisible();
   await expect(page.getByText('Role for e2e test').first()).toBeVisible();
-  await homePage.goToOpenMRSClientAttributes();
+  await homePage.goToClientAttributes();
   await expect(page.getByText('Organizational: Registration Clerk')).toBeTruthy();
   await expect(page.getByText('Application: Edits Existing Encounters')).toBeTruthy();
   await expect(page.getByText('Application: Uses Patient Summary')).toBeTruthy();
```

Comment:
In a case like this
* Replay → navigating to OpenMRS and adding a role in OpenMRS.
* Verification → navigating to Keycloak and making the necessary set of assertions.
* There isn't really any setup beyond whatever was performed in the `beforeEach(..)` routine.

Looks like in many test cases the navigation to component A is considered as part of the setup, and IMO this shouldn't be the case.

Also I see a munch of routines like `homePage.goToKeycloak()` but I don't see a `homePage.goToOpenMRS()`, why is that?

Reply:
> Looks like in many test cases the navigation to component A is considered as part of the setup,

Navigation to component A is a common code across all test-cases and that is why it is considered  in the beforeEach() function.

>  and IMO this shouldn't be the case

Are you suggesting to have the step of navigating to component A to be in each test case? If yes, then I will need to do some refactoring across all the tests. 

>  Also I see a bunch of routines like homePage.goToKeycloak() but I don't see a homePage.goToOpenMRS(), why is that?

We've the method named   `homePage.initiateLogin()`. We would rename the method to  `homePage.goToOpenMRS()` however, the login that is performed is for SSO for Ozone.

Reply:
> If a bunch of tests benefit from gathering a couple of actions in beforeEach(), then do it.

Ok